src/pwm.rs

```diff
@@ -80,6 +82,26 @@ pub trait Pins<REMAP, P> {
     const C3: bool = false;
     const C4: bool = false;
     type Channels;
+
+    fn check_used(c: Channel) -> Channel {
+        if (c == Channel::C1 && Self::C1)
+            || (c == Channel::C2 && Self::C2)
+            || (c == Channel::C3 && Self::C3)
+            || (c == Channel::C4 && Self::C4)
+        {
+            c
+        } else {
+            panic!("Unused channel")
```

Comment:
Still not a fan of this panic, and I'm not sure if check_used is even nessecary anymore

Reply:
@TheZoq2 It took me a bit to understand as this was a @burbull construction, but the function seems to make sense to me.  Without the function there is no way for the per-channel functions to know if that channel is, in-fact, being used.  As an example, lets say that I configure the Timer to use only Channel 3 passing a single pin.  But then I call something like:

```
mypwn.enable(Channel::C1);
```

Without the `check_used` method, the function would go ahead and set the bit.  *Although, I am not sure that are any real negative consequences of this.*

This was the real reason behind me using a macro in my original incarnation of this functionality.  The macro expanded to only include channels that were active and would default to doing nothing through a default match.  In this incantation, this has been shifted to be run-time vice compile-time.

Now, concerning the panic.  Although I had originally default to no action, I actually see no issue with this panic as it will aggressively inform the user that he/she has something amis.  I kind of feel that this is a better approach than doing nothing.  In the do nothing variant, I could mis-type something and work for hours trying to figure out why my channel duty is not acting correctly; only to find out that I had put the wrong channel in there.  If there was a panic, the compiler would yell at me (I think). If I am mis-understanding that it would occur at the compiler phase and it would be runtime, I would have a potential backtrace which would point to the exact place that needed to be checked.

Reply:
> Now, concerning the panic. Although I had originally default to no action, I actually see no issue with this panic as it will aggressively inform the user that he/she has something amis.

Agreed, panic is better than doing nothing

>  If there was a panic, the compiler would yell at me (I think). If I am mis-understanding that it would occur at the compiler phase and it would be runtime,
Panics happen at runtime, if possible I would like to move this to compile time but that is a difficult task.

> If I am mis-understanding that it would occur at the compiler phase and it would be runtime, I would have a potential backtrace which would point to the exact place that needed to be checked.

Yes, but backtraces can be a bit of a pain to get a hold of in embedded contexts

But again, we can probably take care of this in the future in another PR, so unless burbull has additional comments, I think this is good to go

Reply:
I don't know how to do this check compile time because `channel` can be not constant.

In fact we need possibility to write `const fn` in trait implementation and variant of `assert` which fails in compile when it is possible and fail in runtime in other cases.